test.disabled/ejabberd_tests/tests/push_integration_SUITE.erl

```diff
@@ -0,0 +1,553 @@
+            ?assertMatch(#{<<"title">> := AliceJID}, APNSAlert),
+            ?assertMatch(#{<<"badge">> := 1}, APNSData),
+            ?assertMatch(#{<<"category">> := null}, APNSData),
+
+            ok
+        end).
+
+pm_msg_notify_on_fcm_no_click_action(Config) ->
+    escalus:story(
+        Config, [{bob, 1}, {alice, 1}],
+        fun(Bob, Alice) ->
+            PubsubJID = <<"pubsub.localhost">>,
+            Node = {_, NodeName} = pubsub_node(),
+            AliceJID = bare_jid(Alice),
+            DeviceToken = gen_token(),
+
+            pubsub_tools:create_node(Bob, Node, [{type, <<"push">>}]),
+            escalus:send(Bob, enable_stanza(PubsubJID, NodeName,
+                                            [{<<"service">>, <<"fcm">>},
+                                             {<<"device_id">>, DeviceToken}])),
+            escalus:assert(is_result, escalus:wait_for_stanza(Bob)),
+            become_unavailable(Bob),
+
+            escalus:send(Alice, escalus_stanza:chat_to(Bob, <<"OH, HAI!">>)),
+
+            [Notification = #{}] = get_push_logs(fcm, DeviceToken, Config),
+
+            FCMData = maps:get(<<"notification">>, maps:get(<<"request_data">>, Notification)),
+            ?assertMatch(#{<<"body">> := <<"OH, HAI!">>}, FCMData),
+            ?assertMatch(#{<<"title">> := AliceJID}, FCMData),
+            ?assertMatch(#{<<"tag">> := AliceJID}, FCMData),
+            ?assertMatch(#{<<"click_action">> :=  null}, FCMData),
+
+            ok
+        end).
+
+pm_msg_notify_on_apns_w_click_action(Config) ->
+    escalus:story(
+        Config, [{bob, 1}, {alice, 1}],
+        fun(Bob, Alice) ->
+            PubsubJID = <<"pubsub.localhost">>,
+            Node = {_, NodeName} = pubsub_node(),
+            AliceJID = bare_jid(Alice),
+            DeviceToken = gen_token(),
+
+            pubsub_tools:create_node(Bob, Node, [{type, <<"push">>}]),
+            escalus:send(Bob, enable_stanza(PubsubJID, NodeName,
+                                            [{<<"service">>, <<"apns">>},
+                                             {<<"device_id">>, DeviceToken},
+                                             {<<"click_action">>, <<"myactivity">>}])),
+            escalus:assert(is_result, escalus:wait_for_stanza(Bob)),
+            become_unavailable(Bob),
+
+            escalus:send(Alice, escalus_stanza:chat_to(Bob, <<"OH, HAI!">>)),
+
+            [Notification = #{}] = get_push_logs(apns, DeviceToken, Config),
+
+            APNSData = maps:get(<<"aps">>, maps:get(<<"request_data">>, Notification)),
+            APNSAlert = maps:get(<<"alert">>, APNSData),
+            ?assertMatch(#{<<"body">> := <<"OH, HAI!">>}, APNSAlert),
+            ?assertMatch(#{<<"title">> := AliceJID}, APNSAlert),
+            ?assertMatch(#{<<"badge">> := 1}, APNSData),
+            ?assertMatch(#{<<"category">> := <<"myactivity">>}, APNSData),
+
+            ok
+        end).
+
+pm_msg_notify_on_fcm_w_click_action(Config) ->
```

Comment:
Here as well, the duplicated code could be eliminated by reusing some parts of prev test case.